extensions/hibernate-reactive/deployment/src/main/java/io/quarkus/hibernate/reactive/deployment/HibernateReactiveProcessor.java

```diff
@@ -95,6 +98,17 @@ public final class HibernateReactiveProcessor {
             "org.hibernate.reactive.persister.collection.impl.ReactiveBasicCollectionPersister",
     };
 
+    @BuildStep
+    NativeImageFeatureBuildItem registerServicesForReflection(BuildProducer<ServiceProviderBuildItem> services) {
+        services.produce(new ServiceProviderBuildItem(
+                "io.vertx.core.spi.VertxServiceProvider",
+                "org.hibernate.reactive.context.impl.ContextualDataStorage"
+        ));
+
+        // I'm reusing the class in ORM, I don't see anything specific to it
+        return new NativeImageFeatureBuildItem(RegisterServicesForReflectionFeature.class);
```

Comment:
Pretty sure you can just return `void`. This just registers a duplicate feature which most likely will be ignored.

Reply:
Thanks, changed